net-analyzer/monitoring-plugins/files/monitoring-plugins-mariadb-include-fix.patch

```diff
@@ -0,0 +1,40 @@
+--- a/plugins/check_mysql.c	2016-11-29 08:45:08.000000000 +0000
++++ b/plugins/check_mysql.c	2017-10-06 20:55:52.481315949 +0000
+@@ -58,7 +58,7 @@
+ bool ssl = false;
+ char *opt_file = NULL;
+ char *opt_group = NULL;
+-unsigned int db_port = MYSQL_PORT;
++unsigned int db_port = 3306;
```

Comment:
This is incorrect, it should be 0

Reply:
If you then print the help of check_mysql you will get 0 for port. Right now it prints 3306.

Reply:
Assuming 3306 is configured for the server is wrong.  Instead, 0 reads the defaults at runtime from library build time then also local config files like every other client.

Reply:
Well it has worked like this for a long time and let's face it,most mysql instances do run on 3306. However,it's just the default,one can override it.

Reply:
However I'm fine by changing it,just that the help will print port 0 instead of 3306.

Reply:
This isn't a typical MySQL client -- it's performing a check to see whether or not MySQL is running on the expected port. Often, the MySQL instance that you're checking won't be the one running on the local machine. For that reason -- if I were designing the plugin myself -- I would have it default to 3306 rather than whatever is in the local `my.cnf`, since 3306 is the "mysql port" that's defined in `/etc/services` and is the most commonly-used port for MySQL. 

However, doing so would change the behavior of the plugin (which used to use `MYSQL_PORT`). I wouldn't do that unless it could be done upstream.

Reply:
Sorry guys, I overlooked the fact that with MariaDB 10.2.9 the MYSQL_PORT is defined in mysql_version.h. I'll update the PR.